eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -202,12 +207,24 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
           val assistedRoutes1 = handleUpdate(nodeId, failureMessage, d)
           val ignore1 = PaymentFailure.updateIgnored(failure, ignore)
           // let's try again, router will have updated its state
-          router ! RouteRequest(nodeParams.nodeId, c.targetNodeId, c.finalPayload.amount, c.getMaxFee(nodeParams), assistedRoutes1, ignore1, c.routeParams, paymentContext = Some(cfg.paymentContext))
+          c match {
+            case (c: SendPaymentToRoute) =>
+              c.route.fold(
+                hops => router ! FinalizeRoute(c.finalPayload.amount, hops, c.assistedRoutes, paymentContext = Some(cfg.paymentContext)),
```

Comment:
In case the updated channel comes from the invoice and not the public graph, you need to pass it to `FinalizeRoute`:

```suggestion
                hops => router ! FinalizeRoute(c.finalPayload.amount, hops, assistedRoutes1, paymentContext = Some(cfg.paymentContext)),
```